src/new.spec.js

```diff
@@ -252,13 +252,16 @@ describe("withStore", () => {
       const { effects, getParentProps } = makeTestInstance({
         initialState: () => ({ myEntry: "bar" }),
         effects: {
-          myEffect() {
+          async myEffect() {
             assert(isReadOnly(this));
 
             assert(isReadOnly(this.effects));
             expect(ownProps(this.effects)).toEqual(["myEffect", "_setState"]);
 
             expect(typeof this.resetState).toBe("function");
+            this.state.myEntry = "thud";
```

Comment:
It is *necessary* to test the state before `resetState`!

You can move this after the state tests below.